### PR TITLE
Stabilise openshift 4.18 on Ignition 3.4.0 spec

### PR DESCRIPTION
**Pull request by yasminvalim**
Here’s what this PR aims to achieve:

- [x] Stabilise Openshift 4.18
- [x] Uses Ignition 3.4 specs
- [x] Add specs 4.19.0-experimental
- [x] Remove GRUB password support from OCP 4.18

Fixes: https://github.com/coreos/butane/issues/564

### CODE REVIEW

File: config/openshift/v4_18/schema.go [hunk 1]
```diff
@@ -15,7 +15,7 @@
 package v4_18
 
 import (
-	fcos "github.com/coreos/butane/config/fcos/v1_6"
+	fcos "github.com/coreos/butane/config/fcos/v1_5"
```
Comment (travier) [inline config/openshift/v4_18/schema.go:18]: This should be part of the first commit

File: config/openshift/v4_18/translate_test.go [hunk 2]
```diff
@@ -19,9 +19,9 @@ import (
 	"testing"
 
 	baseutil "github.com/coreos/butane/base/util"
-	base "github.com/coreos/butane/base/v0_6"
+	base "github.com/coreos/butane/base/v0_5"
```
Comment (travier) [inline config/openshift/v4_18/translate_test.go:22]: idem

File: config/openshift/v4_19_exp/translate.go [hunk 3]
```diff
@@ -108,11 +108,11 @@ func (c Config) FieldFilters() *cutil.FieldFilters {
 	return &fieldFilters
 }
 
-// ToMachineConfig4_18Unvalidated translates the config to a MachineConfig.  It also
```
Comment (travier) [inline config/openshift/v4_19_exp/translate.go:111]: second commit

File: config/config.go [hunk 4]
```diff
@@ -38,8 +38,9 @@ import (
 	openshift4_14 "github.com/coreos/butane/config/openshift/v4_14"
 	openshift4_15 "github.com/coreos/butane/config/openshift/v4_15"
 	openshift4_16 "github.com/coreos/butane/config/openshift/v4_16"
-	openshift4_17_exp "github.com/coreos/butane/config/openshift/v4_17"
```
Comment (travier) [inline config/config.go:41]: should be split in a first commit

File: internal/doc/main.go [hunk 5]
```diff
@@ -128,7 +128,7 @@ func generate(dir string) error {
 			"openshift",
 			[]version{
 				// inverse order of website navbar
-				{"4.18.0-experimental", openshift4_18_exp.Config{}},
+				{"4.18.0", openshift4_18.Config{}},
```
Comment (travier) [inline internal/doc/main.go]: Duplicated entries fixed in a later commit

**Comment by travier** [comment]
- 1rst commit: Fix the 4.17.0 imports that we missed from a previous PR
- 2nd commit: All the changes needed to stabilize 4.18.0
- 3rd commit: Add 4.19.0-exp
- 4th commit: Rebuild the docs

**Comment by yasminvalim** [comment]
> Commit split that should help review:
> 
> * 1rst commit: Fix the 4.17.0 imports that we missed from a previous PR
> * 2nd commit: All the changes needed to stabilize 4.18.0, dropping down to ignition 3.4.0, fcos 1.5.0, base 0.5
> * 3rd commit: Add 4.19.0-exp, which is copied from 4.18 before the stabilization, and based on latest exp for ignition, fcos & base
> * 4th commit: Rebuild the docs

Thanks for that! I reorganized the commits using this as example and I guess it's easier to review now. Let me know if there is any other changed needed.

**Comment by travier** [comment]
Yes, 4.18 will not have GRUB password support in the MCO so we need to remove it there.

**Comment by yasminvalim** [comment]
> Yes, 4.18 will not have GRUB password support in the MCO so we need to remove it there.

Nice, I added one more commit to remove it. I think we are good now, the commit are properly organized.

**Comment by travier** [comment]
Failing on:
```
Error: config/openshift/v4_18/translate.go:307:6: func `translateUserGrubCfg` is unused (unused)
  func translateUserGrubCfg(config *types.Config, ts *translate.TranslationSet) translate.TranslationSet {
       ^
```

File: config/openshift/v4_18/translate.go [hunk 6]
```diff
@@ -174,11 +174,11 @@ func (c Config) ToMachineConfig4_18(options common.TranslateOptions) (result.Mac
 	return cfg.(result.MachineConfig), r, err
 }
 
-// ToIgn3_6Unvalidated translates the config to an Ignition config.  It also
+// ToIgn63_4Unvalidated translates the config to an Ignition config.  It also
```
Comment (travier) [inline config/openshift/v4_18/translate.go:176]: `ToIgn3_4Unvalidated`

File: config/config.go [hunk 7]
```diff
@@ -39,7 +39,8 @@ import (
 	openshift4_15 "github.com/coreos/butane/config/openshift/v4_15"
 	openshift4_16 "github.com/coreos/butane/config/openshift/v4_16"
 	openshift4_17 "github.com/coreos/butane/config/openshift/v4_17"
-	openshift4_18_exp "github.com/coreos/butane/config/openshift/v4_18_exp"
```
Comment (travier) [inline config/config.go:42]: The 4.18 part should be in the 4.18 commit, the 4.19 one in the 4.19 commit.

File: internal/docs/config-openshift-v4_10.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
+---
```
Comment (travier) [inline internal/docs/config-openshift-v4_10.md:1]: Hum that looks weird?

File: docs/config-openshift-v4_19-exp.md [hunk 9]
```diff
@@ -1,12 +1,12 @@
 ---
 # This file is automatically generated from internal/doc and Ignition's
 # config/doc.  Do not edit.
-title: OpenShift v4.18.0-experimental
```
Comment (travier) [inline docs/config-openshift-v4_19-exp.md:4]: OK, I said to split the docs change in a fourth commit but that makes things hard to review so let's push the doc changes in each commit instead.

**Review by travier — COMMENTED** [review]
Some more splitting to make this easier to review. I'll do the update.

File: internal/doc/main.go [hunk 10]
```diff
@@ -139,6 +139,7 @@ func generate(dir string) error {
 				{"4.15.0", openshift4_15.Config{}},
 				{"4.16.0", openshift4_16.Config{}},
 				{"4.17.0", openshift4_17.Config{}},
+				{"4.18.0", openshift4_17.Config{}},
```
Comment (travier) [inline internal/doc/main.go:142]:
```suggestion
				{"4.18.0", openshift4_18.Config{}},
```

**Review by travier — APPROVED** [review]
This LGTM now. @prestist Can I let you take a final quick look and merge this? Thanks

**Comment by yasminvalim** [comment]
@travier  Thank you!

**Comment by prestist** [comment]
@travier  sure thing :) thank you both for your efforts on this, looking now

**Review by prestist — APPROVED** [review]
LGTM, I dont see any issues. Great work all, Thank you for getting this done 🚀

**Pull request merged**